pkg/csvcopy/transaction.go

```diff
@@ -0,0 +1,182 @@
+func (tr Transaction) Get(ctx context.Context, conn sqlx.QueryerContext) (*TransactionRow, error) {
+	return getTransactionRow(ctx, conn, tr.loc.FileID, tr.loc.StartRow)
+}
+
+func getTransactionRow(ctx context.Context, conn sqlx.QueryerContext, fileID string, startRow int64) (*TransactionRow, error) {
+	row := &TransactionRow{}
+
+	err := conn.QueryRowxContext(ctx, `
+		SELECT file_id, start_row, row_count, byte_offset, byte_len, created_at, state, failure_reason
+		FROM timescaledb_parallel_copy_transactions
+		WHERE file_id = $1 AND start_row = $2
+		LIMIT 1
+	`, fileID, startRow).Scan(
+		&row.FileID, &row.StartRow, &row.RowCount, &row.ByteOffset, &row.ByteLen, &row.CreatedAt, &row.State, &row.FailureReason,
+	)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, nil
+	}
+	if err != nil {
+		return row, err
+	}
+
+	return row, nil
+}
+
+// Next returns the Next transaction in the sequence.
+// If it returns nil, it means there is no Next transaction
+func (tr Transaction) Next(ctx context.Context, conn sqlx.QueryerContext) (*Transaction, *TransactionRow, error) {
+	row := TransactionRow{}
+
+	err := conn.QueryRowxContext(ctx, `
+		SELECT file_id, start_row, row_count, byte_offset, byte_len, created_at, state, failure_reason
+		FROM timescaledb_parallel_copy_transactions
+		WHERE file_id = $1 AND start_row > $2
+		ORDER BY start_row ASC
+		LIMIT 1
+	`, tr.loc.FileID, tr.loc.StartRow).Scan(
+		&row.FileID, &row.StartRow, &row.RowCount, &row.ByteOffset, &row.ByteLen, &row.CreatedAt, &row.State, &row.FailureReason,
+	)
+
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, nil, nil
+	}
+	if err != nil {
+		return nil, nil, err
+	}
+	next := newTransactionAt(Location{
+		FileID:     row.FileID,
+		StartRow:   row.StartRow,
+		RowCount:   row.RowCount,
+		ByteOffset: row.ByteOffset,
+		ByteLen:    row.ByteLen,
+	})
+
+	return next, &row, nil
+}
+
+//go:embed migrations/*
+var migrations embed.FS
+
+const defaultMultiStatementMaxSize = 10 * 1 << 20 // 10 MB
+
+func ensureTransactionTable(connString string) error {
+	dbx, err := connect(connString)
+	if err != nil {
+		return fmt.Errorf("failed to connect to database, %w", err)
+	}
+
+	defer dbx.Close()
+	source, err := iofs.New(migrations, "migrations")
+	if err != nil {
+		return fmt.Errorf("failed to load migration source, %w", err)
+	}
+
+	instance, err := pgx.WithInstance(dbx.DB, &pgx.Config{
+		MigrationsTable:       "timescaledb_parallel_copy_migrations",
+		StatementTimeout:      10 * time.Second,
+		MultiStatementMaxSize: defaultMultiStatementMaxSize,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create target db instance, %w", err)
+	}
+
+	m, err := migrate.NewWithInstance("embed", source, "target_db", instance)
+	if err != nil {
+		return fmt.Errorf("failed to create migrate instance, %w", err)
+	}
+
+	err = m.Up()
+	if err != nil {
+		if errors.Is(err, migrate.ErrNoChange) {
+			return nil
+		}
+		return err
+	}
+	return nil
+}
```

Comment:
I'm not sure about this approach... maybe it is not worth using migrate tool and just a simple create statement will work.
the main problem is that in the future it can be tricky to automatically handle migrations. but maybe we can just guide users when upgrading the tool